library/src/main/java/org/xmtp/android/library/SigningKey.kt

```diff
@@ -19,9 +19,31 @@ import java.util.Date
 interface SigningKey {
     val address: String
 
-    suspend fun sign(data: ByteArray): SignatureOuterClass.Signature?
+    // If this signing key is a smart contract wallet
+    val isSmartContractWallet: Boolean
+        get() = false
 
-    suspend fun sign(message: String): SignatureOuterClass.Signature?
+    // Default chainId value set to 1
+    var chainId: Long
+        get() = 1
+        set(_) {}
+
+    // Default blockNumber value set to null
+    var blockNumber: Long?
+        get() = null
+        set(_) {}
+
+    suspend fun sign(data: ByteArray): SignatureOuterClass.Signature? {
+        throw NotImplementedError("sign(ByteArray) is not implemented.")
+    }
+
+    suspend fun sign(message: String): SignatureOuterClass.Signature? {
+        throw NotImplementedError("sign(String) is not implemented.")
+    }
+
+    suspend fun signSmartContract(message: String): ByteArray {
```

Comment:
Same statement would be okay with calling this `signSCW` if others think that makes it more clear?